libraries/engage/product/helpers/redirects.js

```diff
@@ -0,0 +1,35 @@
+import { isBeta, redirects, getSettings, isAfter } from '@shopgate/engage/core';
+import { ITEM_PATTERN } from '@shopgate/pwa-common-commerce/product/constants';
+import { makeGetProductEffectivityDates } from '../selectors/product';
+
+/**
+ * Enable redirect handler for effectivity dates and more
+ * @returns {void}
+ */
+export const enableRedirectHandler = () => {
+  if (!isBeta()) {
+    return;
+  }
+
+  /** Effectivity dates */
+  const { accessExpired = true } = getSettings('@shopgate/engage/product/EffectivityDates') || {};
```

Comment:
Reading global settings from a widget key has the effect of behaving unexpectedly, when the effectivity dates component is configured differently. I think the config for effectivity dates does not belong to the component.